ibis/expr/datatypes.py

```diff
@@ -366,6 +372,42 @@ def can_implicit_cast(cls, other):
         return isinstance(other, (Floating, Decimal))
 
 
+_VALID_TIMEZONES = frozenset(pytz.all_timezones)
+
+
+@parametric
+class TimestampWithTimezone(DataType):
```

Comment:
Ended up going this route and exposing a lot of missing things regarding sqlalchemy integration. That should be fixed up now.